packages/tools/src/tools/ReferenceCursors.ts

```diff
@@ -0,0 +1,572 @@
+
+    let annotations = getAnnotations(element, this.getToolName());
+
+    if (!annotations?.length) {
+      return renderStatus;
+    }
+
+    //the viewport change from updateStackPosition might not be applied yet, so sometimes the annotation might not be immediately visible
+    annotations = this.filterInteractableAnnotationsForElement(
+      element,
+      annotations
+    ) as Annotations;
+
+    if (!annotations?.length) {
+      return renderStatus;
+    }
+
+    const styleSpecifier: StyleSpecifier = {
+      toolGroupId: this.toolGroupId,
+      toolName: this.getToolName(),
+      viewportId: enabledElement.viewport.id,
+    };
+
+    for (let i = 0; i < annotations.length; i++) {
+      const annotation = annotations[i] as ReferenceCursor;
+      const { annotationUID, data } = annotation;
+      const { handles } = data;
+      const { points } = handles;
+
+      if (!annotationUID) return renderStatus;
+      styleSpecifier.annotationUID = annotationUID;
+
+      const lineWidth = this.getStyle('lineWidth', styleSpecifier, annotation);
+      const lineDash = this.getStyle('lineDash', styleSpecifier, annotation);
+      const color = this.getStyle('color', styleSpecifier, annotation);
+
+      if (points[0].some((e) => isNaN(e))) return renderStatus;
+      const canvasCoordinates = points.map((p) =>
+        viewport.worldToCanvas(p)
+      ) as [Types.Point2];
+
+      // If rendering engine has been destroyed while rendering
+      if (!viewport.getRenderingEngine()) {
+        console.warn('Rendering Engine has been destroyed');
+        return renderStatus;
+      }
+
+      if (!isAnnotationVisible(annotationUID)) {
+        continue;
+      }
+
+      const crosshairUIDs = {
+        upper: 'upper',
+        right: 'right',
+        lower: 'lower',
+        left: 'left',
+      };
+      const [x, y] = canvasCoordinates[0];
+      const centerSpace = 7;
+      const lineLength = 7;
+      drawLine(
+        svgDrawingHelper,
+        annotationUID,
+        crosshairUIDs.upper,
+        [x, y - (centerSpace / 2 + lineLength)],
+        [x, y - centerSpace / 2],
+        { color, lineDash, lineWidth }
+      );
+      drawLine(
+        svgDrawingHelper,
+        annotationUID,
+        crosshairUIDs.lower,
+        [x, y + (centerSpace / 2 + lineLength)],
+        [x, y + centerSpace / 2],
+        { color, lineDash, lineWidth }
+      );
+      drawLine(
+        svgDrawingHelper,
+        annotationUID,
+        crosshairUIDs.right,
+        [x + (centerSpace / 2 + lineLength), y],
+        [x + centerSpace / 2, y],
+        { color, lineDash, lineWidth }
+      );
+      drawLine(
+        svgDrawingHelper,
+        annotationUID,
+        crosshairUIDs.left,
+        [x - (centerSpace / 2 + lineLength), y],
+        [x - centerSpace / 2, y],
+        { color, lineDash, lineWidth }
+      );
+      renderStatus = true;
+    }
+
+    return renderStatus;
+  };
+
+  updateStackPosition(
```

Comment:
This does not represent the underlying logic, `updateStackPosition` makes me think this is only for stackViewport, how about `updateViewportImage` or  `updateViewport`?